### PR TITLE
fix(ast): compute arithmetic negation instead of logical NOT for smol (#278)

### LINKED ISSUE #278: [Bug]: unary negation of a smol (short) computes logical NOT instead of arithmetic negation
### Description

Unary arithmetic negation (`-x`) of a `smol` (short) value computes the
**logical NOT** of the operand instead of its arithmetic negation. Every other
numeric type negates correctly; only `smol` is wrong.

Root cause is a one-character copy/paste error in `handle_unary_expression()`
(`ast.c`), in the `OP_NEG` case:

```c
case OP_NEG:
    if (operand_type == VAR_INT)
    {
        int *result = SAFE_MALLOC(int);
        *result = -(*(int *)operand_value);      // correct: arithmetic negation
        return result;
    }
    else if (operand_type == VAR_SHORT)
    {
        short *result = SAFE_MALLOC(short);
        *result = !(*(short *)operand_value);    // BUG: logical NOT, should be '-'
        return result;
    }
    ...
```

`VAR_INT`, `VAR_FLOAT`, and `VAR_DOUBLE` all use `-`; `VAR_BOOL` correctly uses
`!`. The `VAR_SHORT` arm should use `-`, but uses `!` — so `-x` on a `smol`
collapses to `0` for any non-zero value and `1` for zero.

### Minimal reproduction code

```c
skibidi main {
    smol a = 5;
    yapping("%d", -a);   🚽 prints 0, should print -5

    smol c = 0;
    yapping("%d", -c);   🚽 prints 1, should print 0

    rizz r = 5;
    yapping("%d", -r);   🚽 prints -5 (rizz path is correct)
    bussin 0;
}
```

### Expected behavior

```
-5
0
-5
```

### Actual behavior

```shell
0
1
-5
```

### Brainrot version / commit

dcddf726a3f818c9d4e1eadc040e7a91aa9ba99f

### Operating system

WSL2 (Ubuntu), x86-64

### Additional context

Fix: in the `VAR_SHORT` arm of `OP_NEG` (ast.c ~line 3017), replace
`!(*(short *)operand_value)` with `-(*(short *)operand_value)`.

A `test_cases/*.brainrot` fixture plus an `expected_results.json` entry
covering `smol` negation would prevent regressions (there is currently no test
that negates a `smol`, which is how this slipped through).

### PR DESCRIPTION
**Pull request by Aditya-233**
## Description

Fixes a copy-paste bug in `handle_unary_expression()` (`ast.c`) where the `VAR_SHORT` case under `OP_NEG` used logical NOT (`!`) instead of arithmetic negation (`-`), causing `-x` on a `smol` (short) operand to collapse non-zero values to 0 and 0 to 1.

A regression test fixture `test_cases/unary_negation_smol.brainrot` (and matching expected output in `tests/expected_results.json`) is added to cover:
- Negation of positive `smol` values (`-5`)
- Negation of zero (`0`, verifying it does not collapse to `1`)
- Negation of negative `smol` values (`5`)
- Double negation (`-(-5) = 5`)
- Boundary values (`SHRT_MAX = 32767` negating to `-32767`)
- Storing negated expressions into `smol` variables
- Parity with `rizz` (`int`) negation

## Related Issue

Fixes #278

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Refactor

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have documented my changes in the code or documentation
- [x] I have added tests that prove my change works, at the lowest appropriate layer (`test_cases/*.brainrot` for language-visible behavior; a host-side C test wired into `make test` for internal APIs Brainrot source cannot reach). Required for every bug fix, feature, refactor, performance change, and breaking change — not optional, not "if applicable". See CONTRIBUTING.md ("TESTS ARE MANDATORY").
- [x] Those tests cover the happy path, the original bug (for fixes), error cases, edge cases, **and** adversarial cases. If the existing harness could not reach the behavior, I extended it or added a lower-level test.
- [ ] If this PR cannot affect program behavior (docs/comments/license only), I explained that in the Description instead of skipping the items above in silence
- [x] I have run `make format-check` locally (or `make format` to fix)
- [x] I have run the unit tests locally
- [x] I have run the valgrind memory tests locally
- [x] All new and existing tests pass

### CODE REVIEW

File: ast.c [hunk 1]
```diff
@@ -3986,7 +3986,7 @@ void *handle_unary_expression(ASTNode *node, void *operand_value,
         else if (operand_type == VAR_SHORT)
         {
             short *result = SAFE_MALLOC(short);
-            *result = !(*(short *)operand_value);
+            *result = -(*(short *)operand_value);
```
Comment (leo-aa88) [inline ast.c:3989]: Correct. This was the only `OP_NEG` arm using logical NOT while every other numeric arm used arithmetic negation, so `-x` on a `smol` collapsed non-zero to 0 and 0 to 1. Verified: with the old `!` the checked-in fixture prints `0 1 1 1 0 0 -5`; with this `-` it prints the expected `-5 0 5 5 -32767 -5 -5`. Negation promotes short to int before truncating on store, so `-SHRT_MIN` wraps like the `int` arm rather than trapping under UBSan.

**Review by leo-aa88 — APPROVED** [review]
LGTM.

I built this and verified it end to end rather than eyeballing the one-character change.

**The fix is correct and the scope is exactly right.** The `VAR_SHORT` arm of `OP_NEG` was the only arithmetic-negation arm using `!`; `VAR_INT`/`VAR_FLOAT`/`VAR_DOUBLE` already used `-`, and `VAR_BOOL` correctly keeps `!`. Changing it to `-` brings `smol` into line with the other numeric types. I re-read the whole `OP_NEG` block to confirm no other arm carries the same copy-paste, and none does.

**The fixture is a real regression test, not a rubber stamp.** Running the checked-in `test_cases/unary_negation_smol.brainrot` on this branch prints exactly the expected `-5 / 0 / 5 / 5 / -32767 / -5 / -5`. I then rebuilt the interpreter with the old `!` restored and the same fixture prints `0 / 1 / 1 / 1 / 0 / 0 / -5`, so it fails loudly on the pre-fix code. It covers positive, zero, negative, double negation, the `SHRT_MAX` boundary, assignment into a `smol`, and parity with `rizz`. Expected-output entry in `tests/expected_results.json` matches byte for byte.

**No overflow surprise.** I checked `smol m = -32768; -m;` under the ASan/UBSan build: negation promotes the short to int, negates, then truncates on store, so it wraps to `-32768` with no UBSan trap, identical to how the `int` arm already behaves. Nothing new is introduced here.

Full `pytest` run is green (481 passed, new fixture included), no regression, and `ast.c` is clang-format clean. Ship it.
